src/main/java/com/openshift/internal/restclient/model/build/BuildConfigBuilder.java

```diff
@@ -0,0 +1,176 @@
+import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.images.DockerImageURI;
+import com.openshift.restclient.model.IBuildConfig;
+import com.openshift.restclient.model.IEnvironmentVariable;
+import com.openshift.restclient.model.build.BuildTriggerType;
+import com.openshift.restclient.model.build.IBuildConfigBuilder;
+
+
+public class BuildConfigBuilder implements IBuildConfigBuilder {
+	
+	private BuildConfig bc;
+	private SourceBuildStrategy sourceStrategy;
+	private GitBuildSource source;
+	private String imageStreamTagOutput;
+	private boolean buildOnConfigChange;
+	private boolean buildOnImageChange;
+	private boolean buildOnSourceChange;
+	
+	public BuildConfigBuilder(BuildConfig bc) {
+		this.bc = bc;
+	}
+	
+	@Override
+	public IBuildConfig build() {
+		bc.setBuildStrategy(sourceStrategy);
+		bc.setBuildSource(source);
+
+		//TODO move into bc
+		ModelNode node = bc.getNode();
+		JBossDmrExtentions.set(node, Collections.emptyMap(), "spec.output.to.kind", ResourceKind.IMAGE_STREAM_TAG);
+		JBossDmrExtentions.set(node, Collections.emptyMap(), "spec.output.to.name", imageStreamTagOutput);
+		
+		bc.addBuildTrigger(new WebhookTrigger(BuildTriggerType.GENERIC, UUID.randomUUID().toString(), null));
+		if(buildOnImageChange) {
+			bc.addBuildTrigger(new ImageChangeTrigger(BuildTriggerType.IMAGE_CHANGE, null, null, null));
+		}
+		if(buildOnConfigChange) {
+			bc.addBuildTrigger(new ImageChangeTrigger(BuildTriggerType.CONFIG_CHANGE,null, null, null));
+		}
+		if(buildOnSourceChange) {
+			bc.addBuildTrigger(new WebhookTrigger(BuildTriggerType.GITHUB, UUID.randomUUID().toString(), null));
+		}
+		
+		return bc;
+	}
+	
+	@Override
+	public IBuildConfigBuilder buildOnSourceChange(boolean onSourceChange) {
+		this.buildOnSourceChange = onSourceChange;
+		return this;
+	}
+
+	@Override
+	public IBuildConfigBuilder buildOnImageChange(boolean onImageChange) {
+		this.buildOnImageChange = onImageChange;
+		return this;
+	}
+
+	@Override
+	public IBuildConfigBuilder buildOnConfigChange(boolean onConfigChange) {
+		this.buildOnConfigChange = onConfigChange;
+		return this;
+	}
+
+	@Override
+	public IBuildConfigBuilder toImageStreamTag(String tag) {
+		imageStreamTagOutput = tag;
+		return this;
+	}
+
+	@Override
+	public ISourceStrategyBuilder usingSourceStrategy() {
+		
+		sourceStrategy = new SourceBuildStrategy(new ModelNode(), bc.getPropertyKeys());
+		return new SourceStrategyBuilder(this, sourceStrategy);
+	}
+	
+	@Override
+	public IGitSourceBuilder fromGitSource() {
+		source = new GitBuildSource("", "", "");
+		return new GitSourceBuilder(this, source);
+	}
+
+	class GitSourceBuilder implements IGitSourceBuilder {
+		
+		private IBuildConfigBuilder builder;
+		private GitBuildSource source;
+		
+		GitSourceBuilder(IBuildConfigBuilder builder, GitBuildSource source){
+			this.builder = builder;
+			this.source = source;
+		}
+
+		@Override
+		public IBuildConfigBuilder end() {
```

Comment:
This follows a pattern of what you are able to do with the f8 client.  This pattern is much nicer to work with then the setXXXXX that is required otherwise.  They impl 'end' to imply  you can continue to build up the instance without the final construction.  I would be inclined to change the impl to store local values before actually 'building' the various instance if that would be more agreeable to you.